co-circom/co-circom/src/lib.rs

```diff
@@ -672,11 +676,29 @@ where
                 .insert(name.clone(), parsed_vals.clone());
             shares[2].public_inputs.insert(name.clone(), parsed_vals);
         } else {
-            let [share0, share1, share2] =
-                SerializeableSharedRep3Input::share_rep3(&parsed_vals, &mut rng, seeded, additive);
-            shares[0].shared_inputs.insert(name.clone(), share0);
-            shares[1].shared_inputs.insert(name.clone(), share1);
-            shares[2].shared_inputs.insert(name.clone(), share2);
+            // if all elements are Some, then we can share normally
+            // else we can only share as Vec<Option<T>> and we have to merge unknown inputs later
+            if let Some(parsed_vals) = parsed_vals.iter().copied().collect::<Option<Vec<_>>>() {
```

Comment:
use
parser_vals.iter().all(|x|x.is_some()) 
instead of this. Then you don't have to clone everything

Reply:
you maybe need to flatten the iter also when I think about it when using all

Reply:
> you maybe need to flatten the iter also when I think about it when using all

is already just a Vec<Option<T>>, so no need to